design-system/react/src/components/Dropdown/defs.ts

```diff
@@ -0,0 +1,108 @@
+import type { PopoverProps } from '@radix-ui/react-popover';
+import { createContext, useContext } from 'react';
+import type { AriaButtonProps, AriaMenuOptions } from 'react-aria';
+import type {
+  MenuTriggerState,
+  MenuTriggerProps,
+  ItemProps,
+} from 'react-stately';
+
+import type { ButtonProps } from '../Button';
+import type { MenuItemProps, MenuProps } from '../Menu';
+
+import type { DropdownMenu } from './DropdownMenu';
+import type { DropdownMenuItem } from './DropdownMenuItem';
+import type { DropdownTrigger } from './DropdownTrigger';
+
+import type { Components } from '~/defs';
+import type { CreateComponent } from '~/utils';
+
+// ----------------------------------------------------------------------------
+// Context
+// ----------------------------------------------------------------------------
+
+export type DropdownContext = {
+  triggerRef: React.MutableRefObject<HTMLButtonElement | null>;
+  state: MenuTriggerState;
+  menuTriggerProps: AriaButtonProps<'button'>;
+  menuProps: AriaMenuOptions<unknown>;
+};
+
+export const DropdownCtx = createContext<DropdownContext>(
```

Comment:
remove from here and add it on `Dropdown.tsx` itself since this file should be just for types